### PR TITLE
Create right-side navigation for smaller screen sizes

### LINKED ISSUE #1434: Create right-side navigation for smaller screen sizes
**What needs to be done? Why does it need to be done?**
For the smaller screen sizes, we will be collapsing the list of charts into a right-side navigation that will slide out when the user taps on the dropdown in the top right of the page. This navigation applies to any page that has more than one chart. 

[Mocks →](https://www.figma.com/file/Mw6kSEP2jdr9i7UVEjPuCI/UP-Consolidation?node-id=3420%3A2645) 

**Additional context**
Add any other context about the task here.

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

Created right-side navigation for smaller screen sizes

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #1434 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/core/SectionNavigation/SectionNavigation.tsx [hunk 1]
```diff
@@ -38,7 +43,13 @@ const SectionNavigation: React.FC = () => {
...
   const { sections } = usePageContent(currentPage as PathwaysPage);
 
-  return (
+  const isMultipleSections = enabledSections.length > 1;
+
+  useEffect(() => {
+    setOpen(false);
+  }, [currentSection]);
+
+  const sectionViz = (
```
Comment (jovergaag) [inline src/core/SectionNavigation/SectionNavigation.tsx:52]: Let's call this `sectionLinks`

**Review by jovergaag — CHANGES_REQUESTED** [review]
This looks and works great! There are a few small requests in there around naming, but overall looks amazing, thank you!

File: src/components/Modal/Modal.scss [hunk 2]
```diff
@@ -84,6 +85,7 @@
   }
 
   &__content {
+    overflow-y: overlay !important;
```
[the remark this answers is not in the document]
Comment (nasaownsky) [reply src/components/Modal/Modal.scss:88]: For Firefox support

[the remark this answers is not in the document]
Comment (nasaownsky) [reply src/components/Modal/Modal.scss:88]: On every browser except Firefox it places scrollbar over the content instead of pushing it sideways.

**Review by jovergaag — APPROVED** [review]

**Pull request merged**